tests/virt_autotest/virt_utils.pm

```diff
@@ -302,4 +302,97 @@ sub upload_supportconfig_log {
+    if (get_var('SYSTEM_ROLE', '') eq "kvm") {
+        $cmd .= " | grep -v \'\\-pv-\'";
+    }
+    my $guest_list = script_output $cmd;
+    save_screenshot;
+
+    # mount the remote NFS location of guest assets
+    # OPENQA_URL="localhost" in local openQA instead of the IP, so the line below need to be turned on and set to the webUI IP when you are using local openQA
+    # Tips: Using local openQA, you need "rcnfsserver start & vi /etc/exports; exportfs -r")
+    set_var('OPENQA_URL', "10.67.18.247");
+    my $openqa_server = get_required_var('OPENQA_URL');
+    $openqa_server =~ s/^http:\/\///;
+    my $remote_export_dir = "/var/lib/openqa/factory/other/";
+    my $mount_point       = "/tmp/remote_guest";
+
+    assert_script_run "if [ ! -d $mount_point ];then mkdir -p $mount_point;fi";
+    save_screenshot;
+    # tip: nfs4 is not supported on sles12sp4
+    assert_script_run("mount -t nfs $openqa_server:$remote_export_dir $mount_point", 120);
+    save_screenshot;
+
+    # determine the local location for the guest image file and xml file
+    # have considered of host with sles11sp4 here but no such host currently
+    my $vm_xml_dir       = "/etc/libvirt/";
+    my $installed_os_ver = get_var('VERSION_TO_INSTALL', get_var('VERSION', ''));
+    ($installed_os_ver) = $installed_os_ver =~ /^(\d+)/;
+    if (get_var('SYSTEM_ROLE', '') eq "xen") {
+        $vm_xml_dir .= "libxl/";
+        if ($installed_os_ver eq '11') {
+            $vm_xml_dir = "/etc/xen/vm";
+        }
+        else {
+            $vm_xml_dir = "/etc/libvirt/libxl";
+        }
+    }
+    else {
+        if ($installed_os_ver eq '11') {
+            $vm_xml_dir = "/etc/kvm/vm";
+        }
+        else {
+            $vm_xml_dir = "/etc/libvirt/qemu";
+        }
+    }
+
+    # copy guest images and xml files to local
+    # test aborts if failing in copying all the guests
+    my $remote_guest_count = 0;
+    foreach my $guest (split "\n", $guest_list) {
+        my $guest_asset           = generate_guest_asset_name("$guest");
+        my $remote_guest_xml_file = $guest_asset . '.xml';
+        my $remote_guest_disk     = $guest_asset . '.disk';
+        # change the long guest image name and xml file name to normal ones in local
+        my $rc = script_run("if [ ! -d $vm_xml_dir ];then mkdir -p $vm_xml_dir;fi; cp $mount_point/$remote_guest_xml_file $vm_xml_dir/$guest.xml", 60);
+        save_screenshot;
+        if ($rc ne '0') {
+            record_soft_failure("Fail in copying $mount_point/$remote_guest_xml_file");
+            next;
+        }
+        script_run("ls -l $vm_xml_dir", 10);
+        save_screenshot;
+        my $local_guest_image = script_output "grep '<source file=' $vm_xml_dir/$guest.xml | sed \"s/^\\s*<source file='\\([^']*\\)'.*\$/\\1/\"";
+        $rc = script_run("cp $mount_point/$remote_guest_disk $local_guest_image", 300);    #it took 75 seconds copy from vh016 to vh001
```

Comment:
As this would break the test in case it fails, I'd be  a bit more generous with the time to copy, just in case there's a lot of traffic on the network.

Reply:
tests on OSD will be implemented soon.